src/Framework/Traits.cs

```diff
@@ -128,6 +128,10 @@ public Traits()
         /// Name of environment variables used to enable MSBuild server.
         /// </summary>
         public const string UseMSBuildServerEnvVarName = "MSBUILDUSESERVER";
+        /// <summary>
+        /// Name of environment variables used to use MSBuild server even if non interactive mode has been detected.
+        /// </summary>
+        public const string UseMSBuildServerInNonInteractiveEnvVarName = "MSBUILDUSESERVERINNONINTERACTIVE";
```

Comment:
I'm wondering if, rather than adding a new environment variable here, we should tweak the SDK code to only set MSBUILDUSESERVER if it's outputting to the screen? Then we can just use one env var and not have an awkwardly named variable. (This feels like "MSBUILDREALLYUSESERVER")

Reply:
I guess one usage scenario might be to overwrite the sdk implict opt-out (basically to force use server in noninteractive mode)